### ISSUE TITLE
[Bug]: [group size] Milvus should report error if setting group_size to 0 or negative

### ISSUE THREAD

**Issue by yanliang567** [milvus-io/milvus#36146]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version: master-20240906-21b135c7-amd64
- Deployment mode(standalone or cluster): standalone
- MQ type(rocksmq, pulsar or kafka):    
- SDK version(e.g. pymilvus v2.0.0rc2):pymilvus 2.5.0rc75
```

### Current Behavior

not report error

### Expected Behavior

report an error with meaningful msg

### Steps To Reproduce

```markdown
1. create a collection with some data
2. build index and load
3. search with group by a field, and group_size=-1
```

**Comment by yanliang567** [comment]
BTW, we also need a max value checking for group size.

**Comment by xiaofan-luan** [comment]
let's set min group size to 1 and max to 10

**Comment by xiaofan-luan** [comment]
just some rough numbers

**Comment by yanliang567** [comment]
verified fix on master-20241028-f87acdf2-amd64 and pymilvus-2.5.0rc104
